### ISSUE TITLE
"Failed to run fabric. Please check your settings and try again"

### ISSUE THREAD

**Issue by MatQV** [chasebank87/unofficial-fabric-plugin#7]
Hi, I'm running with this error "Failed to run fabric. Please check your settings and try again", when trying to use the plugin. I have added the needed info, installed the Fabric Connector (and the API server is on), checked the API connection in plugins settings getting a "Fabric connector API Key is valid", so i'm kinda lost.

I'm pretty new with terminal (I'm in MacOS), I can use Fabric in there, but for some reason I get the error I mentioned, so I must be doing something wrong.

Thanks in advance, sorry for the newbie question.

**Comment by chasebank87** [comment]
No worries, can you enable debugging in the unofficial fabric integration settings? Once that’s turned on, go to view developer tools. Open the console and post the error you get?

**Comment by Johannnnes** [comment]
I am having issues as well, since the last update. Before this update everything was working very well. Very cool plugin btw!

Now the API is no longer working properly it seems, at http://127.0.0.1:49152/docs :
This page isn’t working 
|127.0.0.1 sent an invalid response.
ERR_INVALID_HTTP_RESPONSE

I have installed the Fabric Connector and it was working fine before. Now it says all is fine and the API is running, but the page doesn't load.


This is the full error log of Obsidian:

Obsidian Developer Console
plugin:quickadd:19849 Loading QuickAdd
plugin:quickadd:17793 QuickAdd: (LOG) No migrations to run.
plugin:table-editor-obsidian:28854 loading markdown-table-editor plugin
plugin:dataview:20327 Dataview: version 0.5.67 (requires obsidian 0.13.11)
plugin:obsidian-icon-folder:6375 loading iconize
plugin:metadata-menu:25751 +------ Metadata Menu loaded ------x-+
plugin:auto-tag:24281 [autotag][log] AutoTag Logger loaded
plugin:obsidian-charts:80 loading plugin: Charts
plugin:obsidian-chartsview-plugin:826 Loaded Charts View plugin
plugin:obsidian-echarts:96858 loading echarts plugin...
plugin:obsidian-javascript-init:47 Loading JavaScript Init plugin ...
plugin:obsidian-mindmap-nextgen:44515 Loading Mindmap plugin
plugin:codeblock-customizer:23002 loading CodeBlock Customizer plugin
plugin:3d-graph-new:34056 WARNING: Multiple instances of Three.js being imported.
anonymous @ plugin:3d-graph-new:34056
plugin:3d-graph-new:54720 WARNING: Multiple instances of Three.js being imported.
anonymous @ plugin:3d-graph-new:54720
plugin:graph-analysis:36070 loading graph analysis plugin
plugin:juggl:14 Loading Juggl
plugin:multi-properties:2864 t
plugin:mini-toolbar:6 loading MiniToolbar
plugin:terminal:110 [Fabric] Debug mode enabled
plugin:terminal:110 Initialized Excalidraw Image Cache
plugin:terminal:110 Dataview: all 9314 files have been indexed in 4.545s (9314 cached, 0 skipped).
plugin:terminal:110 [Fabric Debug] Fabric logo downloaded successfully
plugin:terminal:110 [Fabric Debug] Registering custom patterns folder watcher
plugin:terminal:110 [Fabric Debug] Watching for deletions in: Fabric/
plugin:terminal:110 [Fabric Debug] Registering custom patterns folder watcher
plugin:terminal:110 [Fabric Debug] Watching for deletions in: Fabric/
app.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getRoot')
    at t.revealLeaf (app.js:1:2273786)
    at FabricPlugin.activateView (plugin:unofficial-fabric-integration:1118:24)
    at HTMLDivElement.eval (plugin:unofficial-fabric-integration:971:12)
127.0.0.1:49152/patterns:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch
    at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30)
    at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16)
    at FabricView.<anonymous> (app.js:1:1440884)
    at app.js:1:237138
    at Object.next (app.js:1:237243)
    at app.js:1:236159
    at new Promise (<anonymous>)
    at v (app.js:1:235904)
    at t.open (app.js:1:1440741)
    at t.<anonymous> (app.js:1:2210563)
    at app.js:1:237138
    at Object.next (app.js:1:237243)
    at a (app.js:1:235961)
127.0.0.1:49152/patterns:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch
    at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30)
    at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16)
    at FabricView.<anonymous> (app.js:1:1440884)
    at app.js:1:237138
    at Object.next (app.js:1:237243)
    at app.js:1:236159
    at new Promise (<anonymous>)
    at v (app.js:1:235904)
    at t.open (app.js:1:1440741)
    at t.<anonymous> (app.js:1:2210563)
    at app.js:1:237138
    at Object.next (app.js:1:237243)
    at a (app.js:1:235961)
app.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getRoot')
    at t.revealLeaf (app.js:1:2273786)
    at FabricPlugin.activateView (plugin:unofficial-fabric-integration:1118:24)
    at HTMLDivElement.eval (plugin:unofficial-fabric-integration:971:12)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch
    at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30)
    at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch
    at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30)
    at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16)
127.0.0.1:49152/patterns:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch
    at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30)
    at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16)
    at FabricView.<anonymous> (app.js:1:1440884)
    at app.js:1:237138
    at Object.next (app.js:1:237243)
    at app.js:1:236159
    at new Promise (<anonymous>)
    at v (app.js:1:235904)
    at t.open (app.js:1:1440741)
    at t.<anonymous> (app.js:1:2210563)
    at app.js:1:237138
    at Object.next (app.js:1:237243)
    at a (app.js:1:235961)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch
    at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30)
    at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 Error testing Fabric Connector API Key: TypeError: Failed to fetch
    at FabricSettingTab.testFabricConnectorApiKey (plugin:unofficial-fabric-integration:1912:30)
    at eval (plugin:unofficial-fabric-integration:1877:18)
    at HTMLButtonElement.<anonymous> (app.js:1:1484574)
eval @ plugin:terminal:110
plugin:graph-analysis:36131 Initialise Graph: 27.539794921875 ms
plugin:terminal:110 Dataview: Dropped cache entries for 32 deleted files.

**Comment by chasebank87** [comment]
Seems like your issue might be on the connector side. Can you see if you have the most recent version from the releases page:Release 1.0.2 · chasebank87/fabric-connectorgithub.comSent from my iPhoneOn Jul 22, 2024, at 5:06 AM, Johannnnes ***@***.***> wrote:﻿
I am having issues as well, since the last update. Before this update everything was working very well. Very cool plugin btw!
Now the API is no longer working properly it seems, at http://127.0.0.1:49152/docs :
This page isn’t working
|127.0.0.1 sent an invalid response.
ERR_INVALID_HTTP_RESPONSE
I have installed the Fabric Connector and it was working fine before. Now it says all is fine and the API is running, but the page doesn't load.
This is the full error log of Obsidian:
Obsidian Developer Console
plugin:quickadd:19849 Loading QuickAdd
plugin:quickadd:17793 QuickAdd: (LOG) No migrations to run.
plugin:table-editor-obsidian:28854 loading markdown-table-editor plugin
plugin:dataview:20327 Dataview: version 0.5.67 (requires obsidian 0.13.11)
plugin:obsidian-icon-folder:6375 loading iconize
plugin:metadata-menu:25751 +------ Metadata Menu loaded ------x-+
plugin:auto-tag:24281 [autotag][log] AutoTag Logger loaded
plugin:obsidian-charts:80 loading plugin: Charts
plugin:obsidian-chartsview-plugin:826 Loaded Charts View plugin
plugin:obsidian-echarts:96858 loading echarts plugin...
plugin:obsidian-javascript-init:47 Loading JavaScript Init plugin ...
plugin:obsidian-mindmap-nextgen:44515 Loading Mindmap plugin
plugin:codeblock-customizer:23002 loading CodeBlock Customizer plugin
plugin:3d-graph-new:34056 WARNING: Multiple instances of Three.js being imported.
anonymous @ plugin:3d-graph-new:34056
plugin:3d-graph-new:54720 WARNING: Multiple instances of Three.js being imported.
anonymous @ plugin:3d-graph-new:54720
plugin:graph-analysis:36070 loading graph analysis plugin
plugin:juggl:14 Loading Juggl
plugin:multi-properties:2864 t
plugin:mini-toolbar:6 loading MiniToolbar
plugin:terminal:110 [Fabric] Debug mode enabled
plugin:terminal:110 Initialized Excalidraw Image Cache
plugin:terminal:110 Dataview: all 9314 files have been indexed in 4.545s (9314 cached, 0 skipped).
plugin:terminal:110 [Fabric Debug] Fabric logo downloaded successfully
plugin:terminal:110 [Fabric Debug] Registering custom patterns folder watcher
plugin:terminal:110 [Fabric Debug] Watching for deletions in: Fabric/
plugin:terminal:110 [Fabric Debug] Registering custom patterns folder watcher
plugin:terminal:110 [Fabric Debug] Watching for deletions in: Fabric/
app.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getRoot')
at t.revealLeaf (app.js:1:2273786)
at FabricPlugin.activateView (plugin:unofficial-fabric-integration:1118:24)
at HTMLDivElement.eval (plugin:unofficial-fabric-integration:971:12)
127.0.0.1:49152/patterns:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch
at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30)
at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16)
at FabricView. (app.js:1:1440884)
at app.js:1:237138
at Object.next (app.js:1:237243)
at app.js:1:236159
at new Promise ()
at v (app.js:1:235904)
at t.open (app.js:1:1440741)
at t. (app.js:1:2210563)
at app.js:1:237138
at Object.next (app.js:1:237243)
at a (app.js:1:235961)
127.0.0.1:49152/patterns:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch
at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30)
at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16)
at FabricView. (app.js:1:1440884)
at app.js:1:237138
at Object.next (app.js:1:237243)
at app.js:1:236159
at new Promise ()
at v (app.js:1:235904)
at t.open (app.js:1:1440741)
at t. (app.js:1:2210563)
at app.js:1:237138
at Object.next (app.js:1:237243)
at a (app.js:1:235961)
app.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getRoot')
at t.revealLeaf (app.js:1:2273786)
at FabricPlugin.activateView (plugin:unofficial-fabric-integration:1118:24)
at HTMLDivElement.eval (plugin:unofficial-fabric-integration:971:12)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch
at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30)
at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch
at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30)
at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16)
127.0.0.1:49152/patterns:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch
at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30)
at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16)
at FabricView. (app.js:1:1440884)
at app.js:1:237138
at Object.next (app.js:1:237243)
at app.js:1:236159
at new Promise ()
at v (app.js:1:235904)
at t.open (app.js:1:1440741)
at t. (app.js:1:2210563)
at app.js:1:237138
at Object.next (app.js:1:237243)
at a (app.js:1:235961)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch
at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30)
at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16)
127.0.0.1:49152/models:1     Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE
plugin:terminal:110 Error testing Fabric Connector API Key: TypeError: Failed to fetch
at FabricSettingTab.testFabricConnectorApiKey (plugin:unofficial-fabric-integration:1912:30)
at eval (plugin:unofficial-fabric-integration:1877:18)
at HTMLButtonElement. (app.js:1:1484574)
eval @ plugin:terminal:110
plugin:graph-analysis:36131 Initialise Graph: 27.539794921875 ms
plugin:terminal:110 Dataview: Dropped cache entries for 32 deleted files.

**Comment by chasebank87** [comment]
Created a separate issue for you, lets work on yours there. let me know if you have the most recent release for the fabric connector:

1.0.2

https://github.com/chasebank87/fabric-connector/releases

> I am having issues as well, since the last update. Before this update everything was working very well. Very cool plugin btw! (using Macbook M1)
> 
> Now the API is no longer working properly it seems, at http://127.0.0.1:49152/docs : This page isn’t working |127.0.0.1 sent an invalid response. ERR_INVALID_HTTP_RESPONSE
> 
> I have installed the Fabric Connector and it was working fine before. Now it says all is fine and the API is running, but the page doesn't load.
> 
> This is the full error log of Obsidian:
> 
> Obsidian Developer Console plugin:quickadd:19849 Loading QuickAdd plugin:quickadd:17793 QuickAdd: (LOG) No migrations to run. plugin:table-editor-obsidian:28854 loading markdown-table-editor plugin plugin:dataview:20327 Dataview: version 0.5.67 (requires obsidian 0.13.11) plugin:obsidian-icon-folder:6375 loading iconize plugin:metadata-menu:25751 +------ Metadata Menu loaded ------x-+ plugin:auto-tag:24281 [autotag][log] AutoTag Logger loaded plugin:obsidian-charts:80 loading plugin: Charts plugin:obsidian-chartsview-plugin:826 Loaded Charts View plugin plugin:obsidian-echarts:96858 loading echarts plugin... plugin:obsidian-javascript-init:47 Loading JavaScript Init plugin ... plugin:obsidian-mindmap-nextgen:44515 Loading Mindmap plugin plugin:codeblock-customizer:23002 loading CodeBlock Customizer plugin plugin:3d-graph-new:34056 WARNING: Multiple instances of Three.js being imported. anonymous @ plugin:3d-graph-new:34056 plugin:3d-graph-new:54720 WARNING: Multiple instances of Three.js being imported. anonymous @ plugin:3d-graph-new:54720 plugin:graph-analysis:36070 loading graph analysis plugin plugin:juggl:14 Loading Juggl plugin:multi-properties:2864 t plugin:mini-toolbar:6 loading MiniToolbar plugin:terminal:110 [Fabric] Debug mode enabled plugin:terminal:110 Initialized Excalidraw Image Cache plugin:terminal:110 Dataview: all 9314 files have been indexed in 4.545s (9314 cached, 0 skipped). plugin:terminal:110 [Fabric Debug] Fabric logo downloaded successfully plugin:terminal:110 [Fabric Debug] Registering custom patterns folder watcher plugin:terminal:110 [Fabric Debug] Watching for deletions in: Fabric/ plugin:terminal:110 [Fabric Debug] Registering custom patterns folder watcher plugin:terminal:110 [Fabric Debug] Watching for deletions in: Fabric/ app.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getRoot') at t.revealLeaf (app.js:1:2273786) at FabricPlugin.activateView (plugin:unofficial-fabric-integration:1118:24) at HTMLDivElement.eval (plugin:unofficial-fabric-integration:971:12) 127.0.0.1:49152/patterns:1 Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30) at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16) at FabricView. (app.js:1:1440884) at app.js:1:237138 at Object.next (app.js:1:237243) at app.js:1:236159 at new Promise () at v (app.js:1:235904) at t.open (app.js:1:1440741) at t. (app.js:1:2210563) at app.js:1:237138 at Object.next (app.js:1:237243) at a (app.js:1:235961) 127.0.0.1:49152/patterns:1 Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30) at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16) at FabricView. (app.js:1:1440884) at app.js:1:237138 at Object.next (app.js:1:237243) at app.js:1:236159 at new Promise () at v (app.js:1:235904) at t.open (app.js:1:1440741) at t. (app.js:1:2210563) at app.js:1:237138 at Object.next (app.js:1:237243) at a (app.js:1:235961) app.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getRoot') at t.revealLeaf (app.js:1:2273786) at FabricPlugin.activateView (plugin:unofficial-fabric-integration:1118:24) at HTMLDivElement.eval (plugin:unofficial-fabric-integration:971:12) 127.0.0.1:49152/models:1 Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30) at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16) 127.0.0.1:49152/models:1 Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30) at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16) 127.0.0.1:49152/patterns:1 Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE plugin:terminal:110 [Fabric Debug] Failed to load patterns from API: TypeError: Failed to fetch at FabricView.loadPatterns (plugin:unofficial-fabric-integration:1648:30) at FabricView.onOpen (plugin:unofficial-fabric-integration:1295:16) at FabricView. (app.js:1:1440884) at app.js:1:237138 at Object.next (app.js:1:237243) at app.js:1:236159 at new Promise () at v (app.js:1:235904) at t.open (app.js:1:1440741) at t. (app.js:1:2210563) at app.js:1:237138 at Object.next (app.js:1:237243) at a (app.js:1:235961) 127.0.0.1:49152/models:1 Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE plugin:terminal:110 [Fabric Debug] Failed to load models from API: TypeError: Failed to fetch at FabricView.loadModels (plugin:unofficial-fabric-integration:1578:30) at FabricView.onOpen (plugin:unofficial-fabric-integration:1296:16) 127.0.0.1:49152/models:1 Failed to load resource: net::ERR_INVALID_HTTP_RESPONSE plugin:terminal:110 Error testing Fabric Connector API Key: TypeError: Failed to fetch at FabricSettingTab.testFabricConnectorApiKey (plugin:unofficial-fabric-integration:1912:30) at eval (plugin:unofficial-fabric-integration:1877:18) at HTMLButtonElement. (app.js:1:1484574) eval @ plugin:terminal:110 plugin:graph-analysis:36131 Initialise Graph: 27.539794921875 ms plugin:terminal:110 Dataview: Dropped cache entries for 32 deleted files.

**Comment by MatQV** [comment]
> No worries, can you enable debugging in the unofficial fabric integration settings? Once that’s turned on, go to view developer tools. Open the console and post the error you get?

Hi mate, sorry for the late reply. I didnt know how to use the developers tools on obsidian, but when you mentioned them I di a little google search and fixed my issue. I had added the FULL PATH to the literal folder I wanted my notes to be saved, instead of adding the name of the "obsidian" folder. I fixed that and everything worked perfectly.

Thanks!
